pkg/controllers/dashboard/cspadaptercharts/controller.go

```diff
@@ -0,0 +1,51 @@
+// Package cspadaptercharts creates the controllers necessary for automatically upgrading the csp adapter chart
+package cspadaptercharts
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/catalogv2/system"
+	"github.com/rancher/rancher/pkg/managedcharts/cspadapter"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/rancher/pkg/wrangler"
+)
+
+func Register(ctx context.Context, wContext *wrangler.Context) error {
+	h := &handler{
+		manager:     wContext.SystemChartsManager,
+		adapterUtil: cspadapter.NewChartUtil(wContext.RESTClientGetter),
+	}
+	wContext.Mgmt.Setting().OnChange(ctx, "csp-adapter-upgrade", h.onSetting)
+
+	return nil
+}
+
+type handler struct {
+	sync.Mutex
+	manager     *system.Manager
+	adapterUtil *cspadapter.ChartUtil
+}
+
+func (h *handler) onSetting(key string, setting *v3.Setting) (*v3.Setting, error) {
+	if setting == nil {
+		return nil, nil
```

Comment:
Question: why not return the setting here just as you do below, inside the if statement? Does the caller of `onSetting` handle the `nil, nil` return values properly?